nirps/qc_checks/check_coordinates.py

```diff
@@ -0,0 +1,137 @@
+        if verbose: print('Position error is too large!')
+        return False
+    else: 
+        if verbose: print('Position error is within tolerance.')
+        return True
+    
+
+# ## Fetching the target's info 
+
+obj_name = 'LHS1140'
+
+MAIN_URL = ('https://docs.google.com/spreadsheets/d/'
+            '1dOogfEwC7wAagjVFdouB1Y1JdF9Eva4uDW6CTZ8x2FM/'
+            'export?format=csv&gid=0')
+PENDING_URL = ('https://docs.google.com/spreadsheets/d/'
+               '1dOogfEwC7wAagjVFdouB1Y1JdF9Eva4uDW6CTZ8x2FM/'
+               'export?format=csv&gid=623506317')
+
+# get main table
+main_request = requests.get(MAIN_URL)
+# open main table
+main_dataframe = pd.read_csv(StringIO(main_request.text))
+# get pending table
+pending_request = requests.get(PENDING_URL)
+# open pending table
+pending_dataframe = pd.read_csv(StringIO(pending_request.text))
+# merge these keeping all rows from main table and adding non-repeating
+#  rows from pending table
+astrom_dataframe = pd.concat([main_dataframe,
+                                pending_dataframe]).drop_duplicates()
+
+# if you prefer an astropy table
+astrom_table = Table.from_pandas(astrom_dataframe)
+
+itarget = np.where(astrom_table['OBJNAME'] == obj_name)[0][0]
+target_table = astrom_table[itarget]
+
+
+# Propogate the star w.r.t its proper motion
+
+def propagate_coords(objdata: Dict[str, Any], obs_time: Time) -> Tuple[SkyCoord, Time]:
+    # deal with distance
+    if objdata['PLX'] <= 0:
+        distance = None
+    else:
+        distance = Distance(parallax=objdata['PLX'] * uu.mas)
+    # need to propagate ra and dec to J2000
+    coords = SkyCoord(ra=objdata['RA_DEG'] * uu.deg,
+                      dec=objdata['DEC_DEG'] * uu.deg,
+                      distance=distance,
+                      pm_ra_cosdec=objdata['PMRA'] * uu.mas / uu.yr,
+                      pm_dec=objdata['PMDE'] * uu.mas / uu.yr,
+                      obstime=Time(objdata['EPOCH'], format='jd'))
+    # work out the delta time between epoch and J2000.0
+    with warnings.catch_warnings(record=True) as _:
+        jepoch = Time(objdata['EPOCH'], format='jd')
+        delta_time = (obs_time.jd - jepoch.jd) * uu.day
+    # get the coordinates
+    with warnings.catch_warnings(record=True) as _:
+        # center the image on the current coordinates
+        curr_coords = coords.apply_space_motion(dt=delta_time)
+        curr_time = obs_time
+    # return some stuff
+    return curr_coords, curr_time
+
+# ## Checking the files of your target
+# Fetch .fits files (TODO: Enter your file path here)
+fits_files = glob.glob('../run_lbl/data_dir_HARPS/science/LHS1140/*.fits')
+
+tolerance = 1/60 # 1 arcmin
+
+success_array = []
+for file in fits_files:
+    time = Time(fits.getheader(file)['DATE-OBS'])
+    coords, time = propagate_coords(target_table, time)
+    success_array.append(coords_check(file, coords.ra.deg, coords.dec.deg, tol = np.arcsin(tolerance), verbose=False))
```

Comment:
be careful with `x = np.arcsin(y)` `np.sin(x)`  it assumes x is in radians, tol seems to be a mix you have `tol = np.arcsin(1/60)`  in the function call and `tol = (1/60)` in the function